src/components/TodoBody/TodoBody.tsx

```diff
@@ -0,0 +1,154 @@
+import classNames from 'classnames';
+import { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  filteringBy: Todo[],
+  newTodoId: number[],
+  tempTodo: Omit<Todo, 'id'> | null,
+  deleteTodo: (todoId: number) => void,
+  updateTodo: (updatedTodo: Todo) => Promise<void>,
+};
+
+export const TodoBody: React.FC<Props> = ({
+  filteringBy,
+  newTodoId,
+  tempTodo,
+  deleteTodo,
+  updateTodo,
+}) => {
+  const [isEditing, setIsEditing] = useState<number | null>(null);
+  const [editValue, setEditValue] = useState('');
+  const editFocus = useRef<HTMLInputElement | null>(null);
+
+  // Activate input OnDoubleclick to edit the todo
+  useEffect(() => {
+    if (editFocus.current) {
+      editFocus.current.focus();
+    }
+  }, [editValue]);
+
+  // Submit the edited todo
+  const handleEditingTodo = (
+    event: React.FormEvent<HTMLFormElement> | null,
+    todo: Todo,
+  ) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    switch (editValue) {
+      case Errors.NULL:
```

Comment:
Comparing with Errors.NULL is not the right way to check if a string is empty. You should directly check if editValue is an empty string.